src/api/mutations/payment/usePostPaymentToStripeMutation.ts

```diff
@@ -0,0 +1,65 @@
+import httpClient from '@/api/httpClient';
+import { ENDPOINTS } from '@/config/endpoints.config';
+import { LOCAL_STORAGE_KEYS } from '@/config/local-storage.config';
+import { ROUTES } from '@/config/routes.config';
+import { ResponsePayment } from '@/types/payment.types';
+import { getValueFromLocalStorage, removeFromLocalStorage } from '@/utils';
+import { CardElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import { useMutation } from '@tanstack/react-query';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+
+function usePostPaymentToStripeMutation(bookingId: string) {
+  const elements = useElements();
+  const stripe = useStripe();
+  const navigate = useNavigate();
+
+  const { mutate: mutatePaymentConfirm } = useMutation({
+    mutationFn: async (status: string) => {
+      const { data } = await httpClient.post<ResponsePayment>(
+        ENDPOINTS.payment.postConfirmPayment,
+        {
+          bookingId,
+          status,
+        }
+      );
+      return data;
+    },
+    onSuccess: (data) => {
+      removeFromLocalStorage(LOCAL_STORAGE_KEYS.clientSecret);
+      toast.success(data.message);
+      navigate(ROUTES.root);
+    },
+  });
+
+  return useMutation({
+    mutationFn: async () => {
+      if (!stripe || !elements) {
+        throw new Error('Unvailable, try again later.');
+      }
+
+      const cardEl = elements.getElement(CardElement);
+
+      const clientSecret = getValueFromLocalStorage<string>(LOCAL_STORAGE_KEYS.clientSecret) || '';
+
+      const { paymentIntent, error } = await stripe.confirmCardPayment(clientSecret, {
+        payment_method: {
+          card: cardEl!,
```

Comment:
I think it is better to handle the null case instead of bypassing `cardEl`

Reply:
From documentation of `getElement`:

`Returns null if no CardElement is rendered in the current Elements provider tree.`

sounds severely to handle, especially in the payment module